app/src/main/java/io/intrepid/contest/base/BaseFragment.kt

```diff
@@ -0,0 +1,147 @@
+package io.intrepid.contest.base
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.annotation.CallSuper
+import android.support.annotation.StringRes
+import android.support.v4.app.Fragment
+import android.support.v7.app.ActionBar
+import android.support.v7.app.AppCompatActivity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+
+import butterknife.ButterKnife
+import butterknife.Unbinder
+import io.intrepid.contest.ContestApplication
+import timber.log.Timber
+
+abstract class BaseFragment<P : BaseContract.Presenter<V>, in V : BaseContract.View> : Fragment(), BaseContract.View {
+
+    protected lateinit var presenter: P
+    private var unbinder: Unbinder? = null
+    private var actionBar: ActionBar? = null
+
+    protected val contestApplication: ContestApplication
+        get() = activity.application as ContestApplication
+
+    @CallSuper
+    override fun onAttach(context: Context?) {
+        Timber.v("Lifecycle onAttach: " + this + " to " + context)
+        super.onAttach(context)
+    }
+
+    @CallSuper
+    override fun onCreate(savedInstanceState: Bundle?) {
+        Timber.v("Lifecycle onCreate: " + this)
+        super.onCreate(savedInstanceState)
+        val configuration = contestApplication.presenterConfiguration
+        presenter = createPresenter(configuration)
+        actionBar = (activity as AppCompatActivity).supportActionBar
+    }
+
+    /**
+     * Override [.onViewCreated] to handle any logic that needs to occur right after inflating the view.
+     * onViewCreated is called immediately after onCreateView
+     */
+    override fun onCreateView(inflater: LayoutInflater?,
+                              container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        Timber.v("Lifecycle onCreateView: " + this)
+        val view = inflater!!.inflate(layoutResourceId, container, false)
```

Comment:
good catch, what it looks like we actually want is to not let LayoutInflater be null (LayoutInflater vs LayoutInflater?). This maps to the kotlin skeleton project better.